base_layer/wallet_ffi/src/lib.rs

```diff
@@ -4199,6 +4263,13 @@ mod test {
             let ffi_completed_txs = wallet_get_completed_transactions(&mut (*alice_wallet), error_ptr);
             assert_eq!(completed_transactions_get_length(ffi_completed_txs, error_ptr), 1);
 
+            let id_completed = completed_transactions_get_at(&mut (*ffi_completed_txs), 0, error_ptr);
+            let id_completed_get =
+                wallet_get_completed_transaction_by_id(&mut (*alice_wallet), (&mut (*id_completed)).tx_id, error_ptr);
+            assert_eq!((*id_completed), (*id_completed_get));
+            completed_transaction_destroy(&mut (*id_completed));
+            completed_transaction_destroy(&mut (*id_completed_get));
+
```

Comment:
I think you should test the cases you added in this PR. Test to see that `wallet_get_pending_outbound_transaction_by_id` and `wallet_get_pending_inbound_transaction_by_id` returns CompletedTransactions in the respective states and directions and check that ` wallet_get_completed_transaction_by_id` does NOT return a completed transaction if it is not in the Mined state.